### ISSUE TITLE
Mappings for multiple contract files

### ISSUE THREAD

**Issue by JoranHonig** [ConsenSys/mythril#550]
Currently a SolidityContract object only gathers/stores mappings for a single source file, in order to do correct issue reporting for multiple file contracts we need to add support for this in the SolidityContract class.

See also #480

**Comment by norhh** [comment]
Yeah, but currently `SolidityContract` class just takes a single file for whatever input right?

**Comment by dmuhs** [comment]
@norhh That's correct. I think what @JoranHonig is getting at here is the limitations resulting from the `SolidityContract` class only being able to hold a single file. That's why we do things [like this](https://github.com/ConsenSys/mythril/blob/master/mythril/mythril.py#L326-L333), where it could actually be handled by the class itself.

It probably would be nice to have all source maps in one place. I'm not sure about the implications for all other parts, though. Will look into that

**Comment by dmuhs** [comment]
So in my understanding, the `SolidityContract` class should hold all input files and their respective source mappings instead of having a list of contract classes on the outside. So assuming a call like `./myth -x test1.sol test2.sol` the `load_from_solidity` function around [here](https://github.com/ConsenSys/mythril/blob/master/mythril/mythril.py#L344) should not return a contract list like
`[<SolidityContract (test1.sol)>, <SolidityContract (test2.sol)>]` but instead a single `SolidityContract` object holding all the files, and their respective `solc` JSON data.

@JoranHonig Please correct me if I misunderstood. I'm also asking myself how that change would improve issue reporting. Unless there's an import hidden in one of the files (in which case it'd be enough to just pass the importing sol file), the both files don't have anything to do with each other and can be treated as separate contracts, can't they?

**Comment by norhh** [comment]
If there are imports then we would pass that path as a solc args for compiling, so there wouldn't be any difference if we treat them as separate contracts as solc will take care of taking in all the imports

**Comment by JoranHonig** [comment]
Let me start out with that I think I made a mistake and what I thought was a problem isn't actually one :sweat_smile: 
On the other hand, here is the weird thing. The class does maintain a list with solidity files, while it only accepts a single solidity_filename as an argument.
